dislib/data/classes.py

```diff
@@ -181,17 +225,14 @@ def _update_labels(self):
 
     def _update_samples(self):
         self.collect()
-        self._samples = np.empty((0, self.n_features))
-
-        for subset in self._subsets:
-            self._samples = np.concatenate((self._samples, subset.samples))
+        if len(self._subsets) > 0:
+            # use the first subset to init to keep the subset's dtype
+            self._samples = self._subsets[0].samples
 
-    def _update_samples_sparse(self):
-        self.collect()
-        self._samples = csr_matrix((0, self.n_features))
+            concat_f = sp.vstack if self._sparse else np.concatenate
 
-        for subset in self._subsets:
-            self._samples = vstack([self._samples, subset.samples])
+            for subset in self._subsets[1:]:
+                self._samples = concat_f((self._samples, subset.samples))
 
```

Comment:
If self._subsets is empty, self._samples is never updated.

Reply:
if self._subsets what do you have to update?